lib/chef/target_io/train/file.rb

```diff
@@ -51,35 +54,37 @@ def open(file_name, mode = "r")
           if block_given?
             yield(io)
 
-            if (content != new_content) && !mode.start_with?("r")
-              __transport_connection.file(file_name).content = new_content # Need Train 2.5+
-            end
+            # Return name of new remote file to be used in later operations
+            file_name = write_file(file_name, new_content) if (content != new_content) && !mode.start_with?("r")
```

Comment:
So this is null in most cases, and then you introduce this def path that returns null, is that... what you expect?

It feels like you'd want:

```ruby
if (content != new_content) && !mode.start_with?("r")
  file_name = write_file(file_name, new_content)
  io.instance_eval <<~IJECT
    def path
      file_name.to_s
    end
  end
end

Reply:
I found a better way. That code was way too convoluted 😅